community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/LogTailScanner.java

```
@@ -147,28 +174,46 @@ private LogTailInformation update() throws IOException
         LogPosition target = latestCheckPoint.getLogPosition();
-        boolean startEntryAfterCheckPoint = latestStartEntry != null &&
-                latestStartEntry.getStartPosition().compareTo( target ) >= 0;
+        boolean startEntryAfterCheckPoint = ((latestStartEntry != null) &&
+                (latestStartEntry.getStartPosition().compareTo( target ) >= 0)) ||
+                commitsAfterCheckPoint;
         if ( !startEntryAfterCheckPoint )
         {
             if ( target.getLogVersion() < version )
             {
                 // This check point entry targets a previous log file.
                 // Go there and see if there's a transaction. Reader is capped to that log version.
-                startEntryAfterCheckPoint = extractFirstTxIdAfterPosition( target, version ) !=
-                        LogTailInformation.NO_TRANSACTION_ID;
+                try
+                {
+                    startEntryAfterCheckPoint =
+                            extractFirstTxIdAfterPosition( target, version ) != LogTailInformation.NO_TRANSACTION_ID;
+                }
+                catch ( LogReadingException lre )
+                {
+                    startEntryAfterCheckPoint = true;
+                }
             }
         }
 
         // Extract first transaction id after check point target position.
         // Reader may continue into log files after the initial version.
-        long firstTxIdAfterCheckPoint = startEntryAfterCheckPoint
-                ? extractFirstTxIdAfterPosition( target, fromVersionBackwards )
-                : LogTailInformation.NO_TRANSACTION_ID;
+        long firstTxIdAfterCheckPoint = LogTailInformation.NO_TRANSACTION_ID;
+        if ( startEntryAfterCheckPoint )
+        {
+            try
+            {
+                firstTxIdAfterCheckPoint = extractFirstTxIdAfterPosition( target, fromVersionBackwards );
+            }
+            catch ( LogReadingException lre )
+            {
+                //
```

Should we silently accept such failure always, or only on "force"? This is changing default behaviour isn't it?